nexus/src/external_api/params.rs

```diff
@@ -284,6 +284,33 @@ pub struct NetworkInterfaceCreate {
     pub ip: Option<IpAddr>,
 }
 
+/// Parameters for updating a
+/// [`NetworkInterface`](omicron_common::api::external::NetworkInterface).
+///
+/// Note that modifying IP addresses for an interface is not yet supported, a
+/// new interface must be created instead.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct NetworkInterfaceUpdate {
+    #[serde(flatten)]
+    pub identity: IdentityMetadataUpdateParams,
+
+    /// Make a secondary interface the instance's primary interface.
+    ///
+    /// If applied to a secondary interface, that interface will become the
+    /// primary on the next reboot of the instance. Note that this may have
+    /// implications for routing between instances, as the new primary interface
+    /// will be on a distinct subnet from the previous primary interface.
+    ///
+    /// Note that this can only be used to select a new primary interface for an
+    /// instance. Requests to change the primary interface into a secondary will
+    /// return an error.
+    // TODO-completeness TODO-docs: When we get there, this should note that a
+    // change in the primary interface will result in changes to the DNS records
+    // for the instance, though not the name.
```

Comment:
But this API will probably remain the same, even when we have external DNS integration, right?

Reply:
Yeah, that's right, I'm referring to the documentation here. This operation is pretty significant in that it completely changes the public networking of an instance, something we'd want to make clear.

Reply:
Just to be clear, a change to the primary nic will not impact the records that are in external DNS, only in VPC-internal DNS. They impact where the external IP's traffic gets 1:1 NAT'd and thus what interface it shows up in the guest, but doesn't change the external DNS with "public" IPs.